ui/src/components/connections/ConnectionDetails.tsx

```diff
@@ -1,40 +1,153 @@
-import { Avatar, Button, Card, Row, Space, Tag, Typography } from "antd";
+import {
+  Avatar,
+  Button,
+  Card,
+  Radio,
+  RadioChangeEvent,
+  Row,
+  Space,
+  Tag,
+  Tooltip,
+  Typography,
+  message,
+} from "antd";
+import Table, { ColumnsType } from "antd/es/table";
 import { useCallback, useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 
-import { Detail } from "../schemas/Detail";
 import { APIError } from "src/adapters/api";
 import { Connection, getConnection } from "src/adapters/api/connections";
-import { ReactComponent as IconCreditCardPlus } from "src/assets/icons/credit-card-plus.svg";
+import {
+  Credential,
+  CredentialType,
+  credentialTypeParser,
+  getCredentials,
+} from "src/adapters/api/credentials";
 import { ReactComponent as IconCreditCardRefresh } from "src/assets/icons/credit-card-refresh.svg";
 import { ReactComponent as IconTrash } from "src/assets/icons/trash-01.svg";
+import { ConnectionDetailsRowDropdown } from "src/components/connections/ConnectionDetailsRowDropdown";
+import { IssueDirectlyButton } from "src/components/connections/IssueDirectlyButton";
+import { Detail } from "src/components/shared/Detail";
 import { ErrorResult } from "src/components/shared/ErrorResult";
 import { LoadingResult } from "src/components/shared/LoadingResult";
+import { NoResults } from "src/components/shared/NoResults";
 import { SiderLayoutContent } from "src/components/shared/SiderLayoutContent";
 import { TableCard } from "src/components/shared/TableCard";
 import { useEnvContext } from "src/contexts/env";
 import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
-import { IDENTIFIER, ISSUE_CREDENTIAL } from "src/utils/constants";
+import {
+  CREDENTIALS,
+  DOTS_DROPDOWN_WIDTH,
+  IDENTIFIER,
+  ISSUE_CREDENTIAL,
+  ISSUE_DATE,
+} from "src/utils/constants";
+import { processZodError } from "src/utils/error";
 import { formatDate } from "src/utils/forms";
-import { AsyncTask, isAsyncTaskDataAvailable, isAsyncTaskStarting } from "src/utils/types";
+import {
+  AsyncTask,
+  hasAsyncTaskFailed,
+  isAsyncTaskDataAvailable,
+  isAsyncTaskStarting,
+} from "src/utils/types";
+
+const obfuscateDID = (did: string) => {
+  const didSplit = did.split(":").slice(0, -1).join(":");
+  const address = did.split(":").pop();
+
+  return address
+    ? `${didSplit}${address.substring(0, 5)}...${address.substring(address.length - 5)}`
+    : "-";
+};
+
+const timeToExpire = (date: Date): string => {
+  const difference = date.getTime() - new Date().getTime();
+
+  if (difference < 0) {
+    return "Expired";
+  }
+
+  const days = Math.floor(difference / (1000 * 60 * 60 * 24));
+  const hours = Math.floor((difference / (1000 * 60 * 60)) % 24);
+  const minutes = Math.floor((difference / (1000 * 60)) % 60);
+
+  if (days > 0) {
+    return `${days}d ${hours}h`;
+  } else if (hours > 0) {
+    return `${hours}h ${minutes}m`;
+  } else {
+    return `${minutes}m`;
+  }
+};
```

Comment:
This implementation produces weird results:

<img width="613" alt="image" src="https://user-images.githubusercontent.com/15896177/229808351-c0f83f93-7454-495b-9ebd-c0c329f17648.png">

I'd better use the plugin from dayjs as discussed in [this thread](https://0xpolygon.slack.com/archives/C0409DDNV7V/p1680614410405649?thread_ts=1678791964.608569&cid=C0409DDNV7V):

```diff
index b7a11db..c6ad0a7 100644
--- a/ui/src/components/connections/ConnectionDetails.tsx
+++ b/ui/src/components/connections/ConnectionDetails.tsx
@@ -12,6 +12,8 @@ import {
   message,
 } from "antd";
 import Table, { ColumnsType } from "antd/es/table";
+import dayjs, { extend as extendDayJsWith } from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
 import { useCallback, useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 
@@ -51,6 +53,8 @@ import {
   isAsyncTaskStarting,
 } from "src/utils/types";
 
+extendDayJsWith(relativeTime);
+
 const obfuscateDID = (did: string) => {
   const didSplit = did.split(":").slice(0, -1).join(":");
   const address = did.split(":").pop();
@@ -60,26 +64,6 @@ const obfuscateDID = (did: string) => {
     : "-";
 };
 
-const timeToExpire = (date: Date): string => {
-  const difference = date.getTime() - new Date().getTime();
-
-  if (difference < 0) {
-    return "Expired";
-  }
-
-  const days = Math.floor(difference / (1000 * 60 * 60 * 24));
-  const hours = Math.floor((difference / (1000 * 60 * 60)) % 24);
-  const minutes = Math.floor((difference / (1000 * 60)) % 60);
-
-  if (days > 0) {
-    return `${days}d ${hours}h`;
-  } else if (hours > 0) {
-    return `${hours}h ${minutes}m`;
-  } else {
-    return `${minutes}m`;
-  }
-};
-
 export function ConnectionDetails() {
   const env = useEnvContext();
 
@@ -120,7 +104,7 @@ export function ConnectionDetails() {
       dataIndex: "expiresAt",
       key: "expiresAt",
       render: (expiresAt: Credential["expiresAt"]) => (
-        <Typography.Text>{expiresAt ? timeToExpire(expiresAt) : "-"}</Typography.Text>
+        <Typography.Text>{expiresAt ? dayjs().to(expiresAt) : "-"}</Typography.Text>
       ),
       sorter: ({ expiresAt: a }, { expiresAt: b }) => {
         if (a && b) {
```

I've asked to @Figmintojas whether we should always show the relative time, no matter whether it's expired or not.

Reply:
I'll add the tooltip and use dayjs just to valid credentials

Reply:
You should use `dayjs().to(expiresAt)` instead of `dayjs(expiresAt).fromNow(true)` so it reads "Expiration in 4 years" instead of "Expiration 4 years".

Reply:
well, let's get @Figmintojas confirmation